### ISSUE TITLE
Bug: [prefer-nullish-coalescing] False positive to `nullable boolean`

### ISSUE THREAD

**Issue by xsjcTony** [typescript-eslint/typescript-eslint#7055]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=CYUwxgNghgTiAEYD2A7AzgF0VNICMAMklMAJYoDmAXPAEZJIQhQrwA+8AriqAGbkhg7eCk4QIAbgCwAKFCRYCZOixgcIAExES5anQZMW0mbOWZ4cNGKwBebLkLEyldhzW4tT3cYD0P+AH2mgF2vFAQuIFRrkEasn5R7vgh8BgwnAjRgWxu6nEyCYFJGincfAJC2bkesvH+gQAqABakaPCoCK3wEJxgANakEACe8ADuSDADLuSpLW1JdYnqeClpGVmBAPybsbUF9QEAgvQAbggdbYrwAPIA0gA0XGi68AAGOa97hQEAQpxYGCaCBQIAAHlgOvAumpOM9pmgrCA2qNSIC3ttPvslg4UmEIpksttdljGnN4GgmkgxEJaAgQCdwpwoBgEBgkG9iq8xkDWLCXii0e82K9HrR-lCMAByS4QUZQIZtV5JPBcgVNdGbTHfeAACRAKDASiaLAoCCFXLZGq5tDgUD6bUBCAgSAopDAjzQ7MdFjECApVIgQgActcGhYQAAHCYYIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC+ILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
declare const case1Loading: boolean | undefined | null;
declare const case2Loading: boolean;

const result = case1Loading || case2Loading;
//    case2  = false        || case2
//    case1  = true         || case2
//    case2  = undefined    || case2

//    This one is luckily working in this case
//    case1  = true         ?? case2

//    Above ones are OK, using `||`

//    But the next one is causing issues with `??`
//    case1  = false        ?? case2
//    This should be evaluate to `case2` when using with `||`, but it's always `case1` with `??`
//    Hence change `||` to `??` breaks the logic, so the rule should NOT report
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/prefer-nullish-coalescing": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

The rule should not report against `nullable boolean`, since it will likely break the logic

### Actual Result

Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator. 11:29 - 11:31
> Fix to nullish coalescing operator (`??`).

### Additional Info

Another solution is to add an `ignoreNullableBoolean` option, since it will likely breaks the logicbreak

**Comment by bradzacher** [comment]
In general we recommend avoiding these sorts of ambiguous logicals and instead explicitly handle the falsey cases separately to make your intent clear.

That being said - this would be covered by the fix in #6487
